### PR TITLE
BRS-161: Expiry EventBridge fix

### PR DESCRIPTION
**Pull request by danieltruong**
### Jira Ticket:
BRS-161

### Jira Ticket URL:
https://bcparksdigital.atlassian.net/browse/BRS-161

### Description:
Fix the cron that auto sets passes to expired. Added docs on pass state.

### CODE REVIEW

**Review by marklise — APPROVED** [review]

**Review by cameronpettit — APPROVED** [review]

**Review by cole — COMMENTED** [review]
This looks like a good change, I like the diagram.

I don't think `formatDate` will work properly on the first though? Which I think means it still won't ever expire passes from the last day of the month.

```
❯ node
Welcome to Node.js v16.2.0.
Type ".help" for more information.
> function formatDate(d) {
... 		  let month = '' + (d.getMonth() + 1),
... 		      day = '' + (d.getDate() - 1), // We need yesterday's date to check for moving active -> expired
... 		      year = d.getFullYear();
...
... 		  if (month.length < 2)
... 		      month = '0' + month;
... 		  if (day.length < 2)
... 		      day = '0' + day;
...
... 		  return [year, month, day].join('-');
... 		}
undefined
> d = new Date(2021, 11, 1)
2021-12-01T08:00:00.000Z
> formatDate(d)
'2021-12-00'
```

**Comment by marklise** [comment]
> let month = '' + (d.getMonth() + 1),
> ... 		      day = '' + (d.getDate() - 1), // We need yesterday's date to check for moving active -> expired
> ... 		      year = d.getFullYear();

Good point - we shouldn't subtract from a formatDate, it should be an ISO date that we subtract a day from, and then convert to a formatted date.  That should make it pass.

**Comment by danieltruong** [comment]
> This looks like a good change, I like the diagram.
> 
> I don't think `formatDate` will work properly on the first though? Which I think means it still won't ever expire passes from the last day of the month.
> 
> ```
> ❯ node
> Welcome to Node.js v16.2.0.
> Type ".help" for more information.
> > function formatDate(d) {
> ... 		  let month = '' + (d.getMonth() + 1),
> ... 		      day = '' + (d.getDate() - 1), // We need yesterday's date to check for moving active -> expired
> ... 		      year = d.getFullYear();
> ...
> ... 		  if (month.length < 2)
> ... 		      month = '0' + month;
> ... 		  if (day.length < 2)
> ... 		      day = '0' + day;
> ...
> ... 		  return [year, month, day].join('-');
> ... 		}
> undefined
> > d = new Date(2021, 11, 1)
> 2021-12-01T08:00:00.000Z
> > formatDate(d)
> '2021-12-00'
> ```

Nice catch. Fixed in f00ff0c.

File: lambda/checkExpiry/index.js [hunk 1]
```diff
@@ -9,7 +9,8 @@ exports.handler = async (event, context) => {
     };
 
     // Look for today's expiries
-    let yesterdaysDate = formatDate(new Date())
+    let yesterdaysDate = new Date();
+    yesterdaysDate.setDate(yesterdaysDate.getDate()-1);
```
Comment (cole) [inline lambda/checkExpiry/index.js:13]: Nice, that's a neat way to do it. I think it needs to be formatted still though.

**Review by cole — CHANGES_REQUESTED** [review]

File: lambda/checkExpiry/index.js [hunk 2]
```diff
@@ -2,62 +2,63 @@ const { runQuery, setStatus } = require('../dynamoUtil');
...
+  }
 }
 
 
 
 function formatDate(d) {
   let month = '' + (d.getMonth() + 1),
-      day = '' + (d.getDate() - 1), // We need yesterday's date to check for moving active -> expired
-      year = d.getFullYear();
+    day = '' + (d.getDate() - 1), // We need yesterday's date to check for moving active -> expired
```
Comment (cole) [inline lambda/checkExpiry/index.js:55]: Remove the -1 here since it's handled on L13 now. Otherwise looks good! 🙌

**Review by cole — APPROVED** [review]

File: lambda/checkExpiry/index.js (hunk 2 above)
Comment (danieltruong) [reply lambda/checkExpiry/index.js]: 2a122fc

**Pull request merged**